framework/src/bundle/BundleContext.cpp

```diff
@@ -197,11 +197,7 @@ std::vector<ServiceReferenceU> BundleContext::GetServiceReferences(
   std::vector<ServiceReferenceU> result;
   std::vector<ServiceReferenceBase> refs;
   b->coreCtx->services.Get(clazz, filter, b, refs);
-  for (std::vector<ServiceReferenceBase>::const_iterator iter = refs.begin();
-       iter != refs.end();
-       ++iter) {
-    result.emplace_back(*iter);
-  }
+  std::copy(refs.begin(), refs.end(), std::back_inserter(result));
```

Comment:
wouldn't it take less cycles to initialize ```result``` using the range constructor instead of copying each element (and potentially having to dynamically grow the ```result``` vector)?